djstripe/models.py

```diff
@@ -547,6 +557,24 @@ def _attach_objects_post_save_hook(self, cls, data):
         # RelatedManager, so this must be done as part of the post save hook.
         cls._stripe_object_to_invoice_items(target_cls=InvoiceItem, data=data, invoice=self)
 
+    @classmethod
+    def upcoming(cls, **kwargs):
+        """ Gets the upcoming preview invoice (singular) for a customer.
+
+        See the documentation in ``djstripe.stripe_objects.StripeInvoice.upcoming()` for details on method parameters.
+        """
+
+        if not issubclass(cls, UpcomingInvoice):
+            if cls is Invoice:
+                cls = UpcomingInvoice
+            else:
+                assert False, "Class argument needs to be derived from UpcomingInvoice"
+
+        stripe_invoice = StripeInvoice.upcoming(**kwargs)
+
+        if stripe_invoice:
+            return cls._create_from_stripe_object(stripe_invoice, save=False)
```

Comment:
Is the class check in 567-571 necessary? You could just use ``UpcomingInvoice`` on this line instead of ``cls``. If that was put in for extendibility, I'd rather go with the class replacement settings idea from a previous conversation.